src/main/java/subway/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,84 @@
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.time.LocalDateTime;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+    public GlobalExceptionHandler() {
+        super();
+    }
+
+    private ResponseEntity<Object> errorResponse(final String ex) {
+        final ErrorResponse errorResponse = ErrorResponse.from(
+                ex,
+                String.valueOf(BAD_REQUEST.value()),
+                LocalDateTime.now());
+
+        return ResponseEntity
+                .status(BAD_REQUEST)
+                .body(errorResponse);
+    }
+
+    private void logInfo(final String message) {
+        log.info(message);
+    }
+
+    private void logError(final String message) {
+        log.error(message);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMissingPathVariable(final MissingPathVariableException ex, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        logInfo(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleTypeMismatch(final TypeMismatchException ex, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        logInfo(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(final MethodArgumentNotValidException ex, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        logInfo(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleBindException(final BindException ex, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        logInfo(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleExceptionInternal(final Exception ex, final Object body, final HttpHeaders headers, final HttpStatus status, final WebRequest request) {
+        logInfo(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
+
+    @ExceptionHandler(value = Exception.class)
+    private ResponseEntity<Object> handleException(final Exception ex) {
+        logError(ex.getMessage());
+        return errorResponse(ex.getMessage());
+    }
```

Comment:
곧장 message 를 반환하면 서버의 중요한 상태 메세지도 사용자에게 노출될 수 있을 것 같아요.
사용자의 다음 행동을 결정지어줄 수 있는 적절한 메세지로 변경해보면 어떨까요?

Reply:
로그는 개발자가 볼 수 있도록 기존 예외 메시지를 남기고
사용자에게 응답되는 메시지만 수정했습니다 !

```suggestion
   @ExceptionHandler(value = Exception.class)
    private ResponseEntity<Object> handleException(final Exception ex) {
        logError(ex.getMessage());
        return errorResponse("예상하지 못한 문제가 발생하였습니다. 관리자에게 문의해주세요.");
    }
```

Reply:
![image](https://github.com/woowacourse/jwp-subway-path/assets/82203978/ce76394d-b661-4709-a757-607b78961b30)

응답 객체 기본 생성자 관련한 예외가 발생해서 넣었습니다 ㅠㅠ
(build and run : IntelliJ)